lib/psych/core_ext.rb

```diff
@@ -17,3 +17,17 @@ def to_yaml options = {}
 if defined?(::IRB)
   require_relative 'y'
 end
+
+
+# TODO: how best to check for builtin Set?
+if defined?(::Set) && Object.const_source_location(:Set) == ["ruby", 0]
+  class Set
+    def encode_with(coder)
+      coder["hash"] = to_h
+    end
+
+    def init_with(coder)
+      replace(coder["hash"].keys)
+    end
+  end
+end
```

Comment:
So I added the implementation, now I'm just not sure what is the best way to check for builtin Set. I could check `RUBY_VERSION >= "3.5"` but I don't know if that may cause issues with JRuby / Truffle ruby?

I could check if `Set#initialize` has a source_location, but then what if someone decorated it?

Etc.

Reply:
You can check if it has a `hash` ivar

Reply:
Good point. `if Set.new.instance_variable_get(:@hash)`, but I think JRuby might not like it because they already move Set into a core class a long time ago.

So perhaps I should check with `Pysch.load(Pysch.dump(Set.new([1, 2])))`. I'll sleep on it.

Reply:
I guess a broader question is: why not move to a more general-purpose implementation completely? Have the behavior change (looks like a set -> doesn't look like a set) after 3.5. Does inter-version consistency matter?

Reply:
Oh, that's an internal implementation